src/dcqc/tests/base_test.py

```diff
@@ -190,16 +206,21 @@ def _find_process_outputs(
             if not path.exists():
                 message = f"Expected process output ({path}) does not exist."
                 raise FileNotFoundError(message)
+
         return outputs
 
     def _interpret_process_outputs(self, outputs: dict[str, Path]) -> TestStatus:
         """Interpret the process output files to yield a test status."""
-        exit_code = outputs["exit_code"].read_text()
-        exit_code = exit_code.strip()
+        exit_code = int(outputs["exit_code"].read_text().strip())
+
         if exit_code == self.pass_code:
             status = TestStatus.PASS
-        else:
+        elif exit_code == self.fail_code:
             status = TestStatus.FAIL
+            self.failure_reason = outputs[self.failure_reason_location].read_text()
```

Comment:
nit: A slightly cleaner way to do:

```
self.failure_reason
and
self.error_reason
```

might be to have just a generic:
```
self.status_reason
```

It is a slight shift in responsibility but I'm thinking about "What if I have a `warning` and I need to return back. Would I need to create a new field named `warning_reason`?".

If you have a generic `status_reason` it'll be "Why is the status the value it is".